comment_iowriter.go

```diff
@@ -56,3 +59,86 @@ func (mc *UnifiedCommentWriter) Post(_ context.Context, c *Comment) error {
 	_, err := fmt.Fprintln(mc.w, s)
 	return err
 }
+
+var _ CommentService = &RDJSONLCommentWriter{}
+
+// RDJSONLCommentWriter
+type RDJSONLCommentWriter struct {
+	w io.Writer
+}
+
+func NewRDJSONLCommentWriter(w io.Writer) *RDJSONLCommentWriter {
+	return &RDJSONLCommentWriter{w: w}
+}
+
+func (cw *RDJSONLCommentWriter) Post(_ context.Context, c *Comment) error {
+	if c.ToolName != "" && c.Result.Diagnostic.GetSource().GetName() == "" {
+		c.Result.Diagnostic.Source = &rdf.Source{
+			Name: c.ToolName,
+		}
+	}
+	b, err := protojson.MarshalOptions{
+		UseProtoNames:     true,
+		UseEnumNumbers:    false,
+		Multiline:         false,
+		EmitUnpopulated:   false,
+		EmitDefaultValues: false,
+	}.Marshal(c.Result.Diagnostic)
+	if err != nil {
+		return err
+	}
+	if _, err = cw.w.Write(b); err != nil {
+		return err
+	}
+	if _, err := cw.w.Write([]byte("\n")); err != nil {
+		return err
+	}
+	return nil
+}
+
+var _ CommentService = &RDJSONCommentWriter{}
+
+// RDJSONCommentWriter
```

Comment:
**[golint-circleci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>comment on exported type RDJSONCommentWriter should be of the form "RDJSONCommentWriter ..." (with optional leading article)
<!-- __reviewdog__:ChA2OWIyM2ViYjM2NGFhNWJlEg9nb2xpbnQtY2lyY2xlY2k= -->